### PR TITLE
105 engagement validation

### PR DESCRIPTION
**Pull request by shawn320**
# Description

- create multiple tags in the enagagement form
- Add validations to the engagement form

Bugs fixed:

- contact not being added when creating engagement (Kevin)
- contact not populating the dropdown on mac (Jordan)
- cannot add comments (Jordan)

## Test Instructions

Engagement form fields Validation:

**Contact Name** - required (select at least one contact)
**Subject** - required
**Engagement Type** - required
**Date** - required, cannot select past date
**Paticipants** - required (minimum of 1 participant)
**Description** - required, max length of 1000 characters
**Comments** - not required, max length of 140 characters
**Tags** - not required, cannot create more than 3 tags, tag name cannot exceed 10 characters, cannot create tag that already exist 

## Help Requested

Got help from Kevin, Marco, and Jordan

## Reviewers

@will0684 @kevinNghiem @bcloutier7 @MarcoGoC @shewood @andr0272 

## Checklist
- [x] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

**Comment by shawn320** [comment]
bug deteted: styling on tags not mobile friendly, tag change sizes when error message appears

File: components/engagement/EngForm.vue [hunk 1]
```diff
@@ -287,35 +319,56 @@ export default {
...
+      timeout: null,
+      attemptSubmit: false
+    }
+  },
+  validations: {
+    engagementDetail: {
+      subject: { required, maxChar: maxLength(50) },
+      type: { required },
+      date: { required },
+      description: { required },
```
Comment (andr0272) [inline components/engagement/EngForm.vue:332]: Description needs `maxlength(1000)` and a validation message.

File: components/engagement/EngForm.vue [hunk 2]
```diff
@@ -287,35 +319,56 @@ export default {
         { type: 'Scrum/Sprint' },
         { type: 'Advisory board/Council Meeting' }
       ],
-      showTag: false
+      showTag: false,
+      timeout: null,
+      attemptSubmit: false
+    }
+  },
+  validations: {
```
Comment (andr0272) [inline components/engagement/EngForm.vue:327]: 'Comments' needs to be added to validation block, with `maxLength(140)`

File: components/engagement/EngForm.vue [hunk 3]
```diff
@@ -326,17 +379,56 @@ export default {
...
-        })
-        this.notification('success', 'engagment created')
-      } catch (e) {
+      this.$v.$touch()
+      this.attemptSubmit = true
+      if (this.$v.$invalid) {
         // eslint-disable-next-line no-console
-        console.log('error : ', e.response)
-        this.notification('error', e.response.data.message)
+        console.log(this.invalidFields)
```
Comment (andr0272) [inline components/engagement/EngForm.vue:417]: Don't need to see this on main.

File: components/engagement/EngForm.vue (hunk 1 above)
Comment (shawn320) [reply components/engagement/EngForm.vue:332]: The text area is locked with maximum of 1000 characters. user cannot enter more than 1000 characters, but I can add the validation message for sure

**Comment by will0684** [comment]
I'm getting this error when creating an engagement:

Screen Shot 2020-07-16 at 10 52 54 AM

**Comment by will0684** [comment]
@shawn320 do you need help fixing the bug I mentioned above?

**Comment by will0684** [comment]
> If you add 3 contacts to the engagement for eg. then delete one, you will be shown form errors at that point. I wasn't expecting to see that at this point.

I am seeing this behaviour as well.

**Comment by shawn320** [comment]
> @shawn320 do you need help fixing the bug I mentioned above?

Thank you, this bug is fixed

**Comment by shawn320** [comment]
> If you add 3 contacts to the engagement for eg. then delete one, you will be shown form errors at that point. I wasn't expecting to see that at this point.

Thank you for point it out, its fixed now.

**Comment by shawn320** [comment]
> Actually yeah if you get the scroll to the error list working right I'll reapprove

OK, Thank you Jordan

File: components/engagement/EngForm.vue [hunk 4]
```diff
@@ -275,54 +335,94 @@ export default {
...
-    },
-    increment() {
-      this.engagementDetail.numParticipants++
-    },
-    decrement() {
-      if (this.engagementDetail.numParticipants > 0) {
-        this.engagementDetail.numParticipants--
-      }
+      this.engagementDetail.contacts = value
+      // eslint-disable-next-line no-console
```
Comment (will0684) [inline components/engagement/EngForm.vue:364]: Don't need this

File: components/engagement/EngForm.vue [hunk 5]
```diff
@@ -275,54 +335,94 @@ export default {
...
-        console.log('error : ', e.response)
-        this.notification('error', e.response.data.message)
+      } else {
+        try {
+          await this.$axios.post('/api/engagement/addEngagement', {
+            engagementDetail
+          })
+          this.notification('success', 'engagment created')
+        } catch (e) {
+          // this.notification('error', e.response.data.message)
```
Comment (will0684) [inline components/engagement/EngForm.vue:424]: This looks like it should be uncommented as it is part of the error handling

**Pull request merged**